From fbe573fa3035b477951ace2892f44f68ee0396a3 Mon Sep 17 00:00:00 2001
From: kelson8 <kelson@kelsoncraft.net>
Date: Tue, 15 Sep 2026 13:34:52 -0400
Subject: [PATCH 160/161] feat: Add infinite sprint toggle to lua

* Fix some values to return 0 in lua_player.cpp.

* Update changelog.
---
 Changelog.md                  |  8 +++++++
 src/extras/lua/lua_player.cpp | 39 ++++++++++++++++++++++++++++-------
 src/extras/lua/lua_player.h   |  1 +
 3 files changed, 40 insertions(+), 8 deletions(-)

diff --git a/Changelog.md b/Changelog.md
index 7117c0fe..8137cf2e 100644
--- a/Changelog.md
+++ b/Changelog.md
@@ -983,6 +983,8 @@ feat: Move version stuff into header
 
 ### Latest 1.2.14-4a
 
+**Commit 1**
+
 feat: Add stubs for setting the zone info in lua
 
 * I have mostly replicated the original scripts for this, although it needs tested and still needs some work.
@@ -992,3 +994,9 @@ feat: Add stubs for setting the zone info in lua
 * Label some more items in the code.
 
 * Update changelog, change version to 1.2.14-4a.
+
+**Commit 2**
+
+feat: Add infinite sprint toggle to lua
+
+* Update changelog.
diff --git a/src/extras/lua/lua_player.cpp b/src/extras/lua/lua_player.cpp
index f66b2c18..c44832a6 100644
--- a/src/extras/lua/lua_player.cpp
+++ b/src/extras/lua/lua_player.cpp
@@ -115,6 +115,10 @@ void RegisterPlayerFunctionsLua(lua_State* L)
     lua_pushcfunction(L, toggleInfiniteHealthLua);
     lua_setfield(L, -2, "set_infinite_health");
 
+	// Toggle infinite sprint.
+	lua_pushcfunction(L, toggleInfiniteSprintLua);
+	lua_setfield(L, -2, "set_infinite_sprint");
+
 	// This works
 	// TODO Get CVector for respawn from file.
     lua_pushcfunction(L, setRespawnPointLua);
@@ -296,7 +300,7 @@ SetPlayerPositionLua(lua_State *L)
 	// Make a random number generator
 	// Make it pick between lists of numbers within the range of the map
 
-	return 1;
+	return 0;
 }
 
 /**
@@ -332,7 +336,7 @@ TogglePlayerControlLua(lua_State *L)
 		CPad::GetPad(0)->SetEnablePlayerControls(PLAYERCONTROL_SHORTCUT_TAXI);
 	}
 
-	return 1;
+	return 0;
 }
 
 //-------
@@ -358,7 +362,7 @@ toggleNeverWantedLua(lua_State *L)
 
 	PlayerCheats::SetNeverWanted(togglePolice);
 
-	return 1;
+	return 0;
 }
 
 // int 
@@ -376,10 +380,29 @@ int toggleInfiniteHealthLua(lua_State *L)
 		return luaL_error(L, "player.set_infinite_health boolean required! Value can either be true or false.");
 	}
 
-
 	bool toggleInfHealth = static_cast<float>(lua_toboolean(L, 1));
 	PlayerFunctions::SetInfiniteHealth(toggleInfHealth);
-	return 1;
+	return 0;
+}
+
+/**
+ * Toggle for infinite sprint.
+ * 
+ * Usage:
+ * player.set_infinite_sprint(true)
+ */
+int
+toggleInfiniteSprintLua(lua_State *L)
+{
+	if(!lua_isboolean(L, 1))
+	{
+		return luaL_error(L, "player.set_infinite_sprint boolean required! Value can either be true or false.");
+	}
+
+	bool toggleInfSprint = static_cast<float>(lua_toboolean(L, 1));
+	PlayerFunctions::SetInfiniteSprint(toggleInfSprint);
+
+	return 0;
 }
 
 /**
@@ -396,7 +419,7 @@ HealPlayerLua(lua_State *L)
 
 	LogFunctions::LogInfo(Defines::luaLogFile, "Player was healed");
 
-	return 1;
+	return 0;
 }
 
 /**
@@ -431,7 +454,7 @@ int setPlayerHealthLua(lua_State *L)
 		FindPlayerPed()->m_fHealth = newHealth;
 	}
 
-	return 1;
+	return 0;
 }
 
 /**
@@ -463,7 +486,7 @@ int setPlayerArmorLua(lua_State *L)
 		FindPlayerPed()->m_fArmour = newArmor;
 	}
 
-	return 1;
+	return 0;
 }
 
 //------- Vehicle --------//
diff --git a/src/extras/lua/lua_player.h b/src/extras/lua/lua_player.h
index 45f3407c..6b6ea370 100644
--- a/src/extras/lua/lua_player.h
+++ b/src/extras/lua/lua_player.h
@@ -37,6 +37,7 @@ extern "C" {
 	// 
 	extern int toggleNeverWantedLua(lua_State *L);
 	extern int toggleInfiniteHealthLua(lua_State *L);
+	extern int toggleInfiniteSprintLua(lua_State *L);
 
 	extern int changePlayerClothesLua(lua_State *L);
 
-- 
2.43.0

